### ISSUE TITLE
Unicode character breaks email template editor

### ISSUE THREAD

**Issue by zakdma** [magento/magento2#39176]
### Preconditions and environment

- Latest Magento version

### Steps to reproduce

1. Go to admin area.
2. Open Marketing/Communications/Email Templates list
3. Click Add New Template button on the top-right corner
4. See email template editor opened
5. Select Magento_Sales/New Order template as a value for Template form field
6. Click Load Template button
7. Enter any name to Template Name field
8. Add unicode symbol U+1F447 (👇) at any place in Template Content editor. Screenshot: https://prnt.sc/bDnJ5BzxSQTM
9. Click Save Template button to save template
10. Open the saved template again to edit

### Expected result

Template Content field contains all template text as saved previously

### Actual result

Template Content field contains text only prior to the inserted unicode character 👇
All further text is truncated
Screenshot: https://prnt.sc/zF5HOkJokwDj

### Triage and priority

- [X] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by engcom-Bravo** [comment]
Hi @zakdma,

Thanks for your reporting and collaboration.

We have verified the issue in Latest 2.4-develop instance and the issue is reproducible.Kindly refer the screenshots.


<img width="1434" alt="Screenshot 2024-09-18 at 15 08 38" src="https://github.com/user-attachments/assets/21739b71-ce38-4fc5-9bb3-4bfe96c3ed7b">


Template Content field contains text only prior to the inserted unicode character 👇 All further text is truncated.

Hence Confirming the issue.

Thanks.

**Comment by hostep** [comment]
Would https://github.com/magento/magento2/pull/35174 fix this?

**Comment by zakdma** [comment]
@magento give me 2.4-develop instance

**Comment by zakdma** [comment]
> Would #35174 fix this?

I don't think so. I tried manually changing the collation for the table and its fields, but it still doesn't work. It stopped trimming strings by such symbols but replaced them with ???? instead.
